libs/utils/env.py

```diff
@@ -474,19 +485,36 @@ def _init_target(self, force = False):
             # Brackets are there to let the output dir be created automatically
                                     '--workload-automation-vio={}'])
 
+            # Add variant suffix
+            suffix = os.path.splitext(os.path.basename(
+                     system_info['platform']['description']))[0]
+            board_file = self.conf['board'].lower() + suffix + '.json';
+            self._board_path = os.path.join(basepath, 'libs/utils/platforms',
+                                      board_file)
+            if not os.path.exists(self._board_path):
+                self._log.warning('Board descriptor file %s was not found.',
+                                  self._board_path)
+                self._board_path = None
+                board = None
+            else:
+                board = self._load_board()
+
             # Merge all arguments
             platform = devlib.platform.gem5.Gem5SimulationPlatform(
-                name='gem5',
-                gem5_bin=simulator_info['gem5_bin'],
-                gem5_args=args,
-                gem5_virtio=virtio_args,
-                host_output_dir=self.res_dir
+                name = 'gem5',
+                gem5_bin = simulator_info['gem5_bin'],
+                gem5_args = args,
+                gem5_virtio = virtio_args,
+                host_output_dir = self.res_dir,
+                core_names = board['cores'] if board else None,
```

Comment:
For a physical platform I would try to reduce as much as possible the amount of required data for the sake of testing simplicity, but since gem5 simulations can be quite lengthy I think it's a good idea to enforce certain parameters before starting a simulation. To me it's good for merging.